jupyter_firefly_extensions/__init__.py

```diff
@@ -36,13 +37,41 @@ def initialize(self, notebook_dir, firefly_url):
         self.firefly_url = firefly_url
 
     def get(self):
+        log = logging.getLogger(__name__)
         path = self.get_argument('path', 'not found')
-        print('sendToFirefly:uploading: ' + path)
         fc = FireflyClient(url=self.firefly_url)
-        upload_name = fc.upload_file(self.notebook_dir+'/'+path)
-        print('sendToFirefly:done: ')
+
+        file_names = self.generate_file_names(path,self.notebook_dir)
+        found = False
+        for f in file_names:
+            if self.can_read(f):
+                log.info('sendToFirefly: uploading: ' + f)
+                upload_name = fc.upload_file(f)
+                found = True
+                break
+
+        if found:
+            log.info('sendToFirefly: success, Upload key: ' + upload_name)
+        else:
+            all_file_str = ', '.join(n for n in filenames)
+            upload_name = 'FAILED: ' + all_file_str
+            log.info('sendToFirefly:failed: could not find file. tried: : ' + all_file_str)
+
         self.finish(upload_name)
 
+    @staticmethod
+    def can_read(f):
+        if os.path.exists(f):
+                return os.access(f, os.R_OK)
+        return false
+
+    @staticmethod
+    def generate_file_names(path, notebook_dir):
+        names = [notebook_dir+'/'+path, os.path.abspath(path)]
```

Comment:
Maybe consider the more generic `os.path.join()`?

Reply:
ok